filesystem/fs_interface.go

```diff
@@ -0,0 +1,13 @@
+package filesystem
+
+type FileSystem interface {
```

Comment:
I mean we need to learn from others, not create our own from the very beginning.